src/main/java/org/gridsuite/shortcircuit/server/utils/SpecificationUtils.java

```diff
@@ -0,0 +1,161 @@
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.query.EscapeCharacter;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+import static org.springframework.data.jpa.domain.Specification.anyOf;
+import static org.springframework.data.jpa.domain.Specification.not;
+
+/**
+ * Utility class to create Spring Data JPA Specification (Spring interface for JPA Criteria API).
+ *
+ * @author Florent MILLOT <florent.millot@rte-france.com>
+ */
+public final class SpecificationUtils {
+
+    // Utility class, so no constructor
+    private SpecificationUtils() {
+    }
+
+    // we use .as(String.class) to be able to works on enum fields
+    public static <X> Specification<X> equals(String field, String value) {
+        return (root, cq, cb) -> cb.equal(getColumnPath(root, field).as(String.class), value);
+    }
+
+    public static <X> Specification<X> contains(String field, String value) {
+        return (root, cq, cb) -> cb.like(getColumnPath(root, field), "%" + EscapeCharacter.DEFAULT.escape(value) + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
+    }
+
+    public static <X> Specification<X> startsWith(String field, String value) {
+        return (root, cq, cb) -> cb.like(getColumnPath(root, field), EscapeCharacter.DEFAULT.escape(value) + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
+    }
+
+    public static <X> Specification<X> notEqual(String field, Double value) {
+        return (root, cq, cb) -> cb.notEqual(getColumnPath(root, field), value);
+    }
+
+    public static <X> Specification<X> lessThanOrEqual(String field, Double value) {
+        return (root, cq, cb) -> cb.lessThanOrEqualTo(getColumnPath(root, field), value);
+    }
+
+    public static <X> Specification<X> greaterThanOrEqual(String field, Double value) {
+        return (root, cq, cb) -> cb.greaterThanOrEqualTo(getColumnPath(root, field), value);
+    }
+
+    public static <X> Specification<X> isNotEmpty(String field) {
+        return (root, cq, cb) -> cb.isNotEmpty(getColumnPath(root, field));
+    }
+
+    public static <X> Specification<X> appendFiltersToSpecification(Specification<X> specification, List<ResourceFilter> resourceFilters) {
+        Objects.requireNonNull(specification);
+
+        if (resourceFilters == null || resourceFilters.isEmpty()) {
+            return specification;
+        }
+
+        Specification<X> completedSpecification = specification;
+
+        for (ResourceFilter resourceFilter : resourceFilters) {
+            if (resourceFilter.dataType() == ResourceFilter.DataType.TEXT) {
+                completedSpecification = appendTextFilterToSpecification(completedSpecification, resourceFilter);
+            }
+            if (resourceFilter.dataType() == ResourceFilter.DataType.NUMBER) {
+                completedSpecification = appendNumberFilterToSpecification(completedSpecification, resourceFilter);
+            }
```

Comment:
use `else if` or switch to not test unnecessary cases
```suggestion
            switch(resourceFilter.dataType()) {
                case TEXT -> completedSpecification = appendTextFilterToSpecification(completedSpecification, resourceFilter);
                case NUMBER -> completedSpecification = appendNumberFilterToSpecification(completedSpecification, resourceFilter);
            }
```

Reply:
switch without default will do a Sonar smell so I just add `else if`, it's OK for 2 values